pkg/build/controller/controller.go

```diff
@@ -228,3 +229,71 @@ func (bc *BuildPodController) CancelBuild(build *buildapi.Build, pod *kapi.Pod)
 func isBuildCancellable(build *buildapi.Build) bool {
 	return build.Status == buildapi.BuildStatusNew || build.Status == buildapi.BuildStatusPending || build.Status == buildapi.BuildStatusRunning
 }
+
+// BuildPodDeleteController watches pods running builds and updates the build if the pod is deleted
+type BuildPodDeleteController struct {
+	BuildStore   cache.Store
+	BuildUpdater buildclient.BuildUpdater
+}
+
+func (bc *BuildPodDeleteController) HandleBuildPodDeletion(pod *kapi.Pod) error {
+	glog.V(4).Infof("Handling deletion of build pod %s/%s", pod.Namespace, pod.Name)
+	var build *buildapi.Build
+	for _, obj := range bc.BuildStore.List() {
```

Comment:
i doubt it, but this is the same pattern we applied here:
https://github.com/openshift/origin/blob/master/pkg/build/controller/controller.go#L147

you're concerned that we're updating the build object in the store?  i'm not really sure it matters if we do, since the store is going to get updated w/ what we're sending to etcd.